### ISSUE TITLE
hardlinks instead of symlinks for the results folder

### ISSUE THREAD

**Issue by cbedetti** [scilus/tractoflow#8]
The `work` directory should not be removed, because of [symlinks](https://tractoflow-documentation.readthedocs.io/en/latest/pipeline/results.html) but may take up a lot of space.

One solution could be to use [hardlinks](https://www.nextflow.io/docs/latest/process.html#publishdir) and add the directive on how to do it in the documentation.

Is it a good idea or deleting the `work` folder is really bad practice?

**Comment by jchoude** [comment]
I personally don't recommend hardlinks, because it sometimes makes things hard to follow, in the event you delete some parts of the work directory.

If you delete the work directory and use hardlinks, you will still use the same space as before deleting. The only change is that you might save some space if unused files were still present in the work directory, but this shouldn't represent a large part of your space use. If work directory space use is a concern, you should try the `nextflow clean` command, to remove everything that is unneeded. 

Also, a note: normally, only files that are kept in the published results are also kept in the work directory. Intermediate files are deleted from the work directory once the process has finished running.

Deleting the work directory is normally not recommended if you intend on rerunning the pipeline with `-resume` (for example, to adjust some parameters). The caching mechanism is entirely dependent on the work directory content.

So, all in all, I wouldn't recommend it. If the issue is that a run of the pipeline uses lots of space, it might be inevitable, since dMRI processing is quite space intensive.

**Comment by cbedetti** [comment]
That makes a lot of sense.
Also, I was not aware of `nextflow clean`, thanks.

I'm closing this issue.
